blitz.mod/blitz.bmx

```diff
@@ -715,103 +755,113 @@ keyword: "Const"
 End Rem
 
 Rem
-bbdoc: Declare a Local variable
+bbdoc: Declare a local variable
 keyword: "Local"
 End Rem
 
 Rem
-bbdoc: Declare a Global variable
+bbdoc: Declare a global variable
 keyword: "Global"
 End Rem
 
 Rem
-bbdoc: Declare a Field variable
+bbdoc: Declare a field variable
 keyword: "Field"
 End Rem
 
 Rem
-bbdoc: Begin a Function declaration
+bbdoc: Begin a function declaration
 keyword: "Function"
 End Rem
 
 Rem
-bbdoc: End a Function declaration
+bbdoc: End a function declaration
 keyword: "EndFunction"
 End Rem
 
 Rem
-bbdoc: Return from a Function
+bbdoc: Begin a method declaration
+keyword: "Method"
+End Rem
+
+Rem
+bbdoc: End a method declaration
+keyword: "EndMethod"
+End Rem
+
+Rem
+bbdoc: Return from a method or function
 keyword: "Return"
 End Rem
 
 Rem
-bbdoc: Begin a user defined Type declaration
+bbdoc: Begin a user defined class declaration
 keyword: "Type"
 End Rem
 
 Rem
-bbdoc: End a user defined Type declaration
+bbdoc: End a user defined class declaration
 keyword: "EndType"
 End Rem
 
 Rem
-bbdoc: Begin a user defined Struct declaration
-keyword: "Struct"
+bbdoc: Begin a user defined interface declaration
+keyword: "Interface"
 End Rem
 
 Rem
-bbdoc: End a user defined Struct declaration
-keyword: "EndStruct"
+bbdoc: End a user defined interface declaration
+keyword: "EndInterface"
 End Rem
 
 Rem
-bbdoc: Specify user defined Type supertype
-keyword: "Extends"
+bbdoc: Begin a user defined structure declaration
+keyword: "Struct"
 End Rem
 
 Rem
-bbdoc: Specify user defined Interface(s)
-keyword: "Implements"
+bbdoc: End a user defined structure declaration
+keyword: "EndStruct"
 End Rem
 
 Rem
-bbdoc: Begin a Method declaration
-keyword: "Method"
+bbdoc: Specify supertype(s) of a user defined type
+keyword: "Extends"
 End Rem
 
 Rem
-bbdoc: End a Method declaration
-keyword: "EndMethod"
+bbdoc: Specify implemented interface(s) of a user defined type
+keyword: "Implements"
 End Rem
 
 Rem
-bbdoc: Denote a Type or Method as Abstract
+bbdoc: Denote a class, function or method as abstract
 keyword: "Abstract"
 End Rem
 
 Rem
-bbdoc: Denote a Type or Method as Final
+bbdoc: Denote a class, function or method as final

```

Comment:
So here you replace "Type" with "class". Some lines later you just replace "Type" with its lowercase variant "type".

Is there  a reason to call it "class" in the lines above?

Reply:
Yes. In BlitzMax, the keyword `Type` is used to define classes. This has historical reasons: BlitzBasic already used the `Type` keyword but didn't have classes, it was only possible to define simple structures. BlitzMax then "upgraded" them to to proper classes with methods, inheritance etc., but the `Type` keyword stayed. (It's very similar to C++ where the `struct` keyword inherited from C can also be used to define classes.)

Now, this can be somewhat confusing, because while classes are indeed types, not all types are classes. `Int or Byte Ptr` for example are builtin primitive types that are not classes, and `Struct`s are user-defined types that are not classes.

This is why I think it's best to use these terms consequently in a manner that is consistent with how they're used in other programming languages:
- When the documentation says "class", it stricly means actual classes (which are unfortunately declared with the `Type` keyword in BlitzMax).
- When the documentation says "user defined type", it means classes, structures, or interfaces.
- Ehen it says just "type", it might mean any type, including primitives.

classes.only all confus for historical reasons

Reply:
Thanks for the elaborative answer. I knew about the ambiguity of "type" but got a bit puzzled when reading some changes of Type->type but also Type->class. It makes sense though.

Danke Dir.